package rocksdb.rpc;

option cc_generic_services = true;

message GetRequest {
    optional string key = 1;
}

message GetResponse {
    optional bool succeed = 1;
    optional string value = 2;
    optional string error = 3;
}

message PutRequest {
    optional string key = 1;
    optional string value = 2;
}

message PutResponse {
    optional bool succeed = 1;
    optional string error = 2;
}

message DeleteRequest {
    optional string key = 1;
}

message DeleteResponse {
    optional bool succeed = 1;
    optional string error = 2;
}

service RocksDB {
    rpc Get(GetRequest) returns(GetResponse);
    rpc Put(PutRequest) returns(PutResponse);
    rpc Delete(DeleteRequest) returns(DeleteResponse);
}
